code/hsec-tools/hsec-tools.cabal

```diff
@@ -55,14 +63,16 @@ library
     , lucid                 >=2.9.0
     , mtl                   >=2.2      && <2.4
     , osv
+    , pandoc                >=2.0      && <3.2
     , pandoc-types          >=1.22     && <2
     , parsec                >=3        && <4
     , pathwalk              >=0.3
+    , pretty                >=1.0      && <1.2
     , process               >=1.6      && <1.7
     , safe                  >=0.3
     , text                  >=1.2      && <3
     , time                  >=1.9      && <1.14
-    , toml-parser           ^>=2.0.0.0
+    , tomland               >=1.3      && <1.4
```

Comment:
now it's bidirectional but parses wrongly and doesn't support the format?

Reply:
I'm a bit partial to not using that specific library but I guess I could not justify this well enough. sorry for that.

Reply:
I understand *toml-parser* provides printing support.  This was one of the advantages @glguy mentioned when proposing we switch to *toml-parser* from *toml-reader*.

Let's look into that before switching up the deps and changing all the code, yet again :)

Reply:
I'm happy to help with generating TOML.

Reply:
@glguy can you provide any guidance to generate a valid Toml file?

Because it's like `prettyToml` is not really doing that.

> now it's bidirectional but parses wrongly and doesn't support the format?

IIUC, the difference between the current implementation and the official specification are limited to new syntax (mostly).

I'm not sure it's an issue since all the advisories are still passing.

On another hand, having to write a distinct writer is error prone and time consuming.

Reply:
You'll have to be more specific about the issue you're having with prettyToml. It certainly generate valid TOML in general.

Let me know that trouble you've found.

Reply:
ok, I have figured it out (modulo one or two changes), thanks everyone